spec/views/hyrax/admin/admin_sets/_show_actions.html.erb_spec.rb

```diff
@@ -0,0 +1,63 @@
+RSpec.describe 'hyrax/admin/admin_sets/_show_actions.html.erb', type: :view do
+  let(:solr_document) { SolrDocument.new }
+  let(:ability) { instance_double("Ability") }
+  let(:presenter) { Hyrax::AdminSetPresenter.new(solr_document, ability) }
+
+  before do
+    allow(controller).to receive(:current_ability).and_return(ability)
+    allow(view).to receive(:presenter).and_return(presenter)
+
+    # Stub route because view specs don't handle engine routes
+    allow(view).to receive(:edit_admin_admin_set_path).with(presenter).and_return("/admin/admin_sets/123/edit")
+    allow(view).to receive(:admin_admin_set_path).with(presenter).and_return("/admin/admin_sets/123")
+  end
+
+  context 'when editor of the admin_set' do
+    before do
+      allow(ability).to receive(:can?).with(:edit, solr_document).and_return(true)
+    end
+
+    context "when presenter has delete disabled" do
+      before do
+        allow(presenter).to receive(:disable_delete?).and_return(true)
+        render
+      end
+      it "displays a disabled delete button" do
+        expect(rendered).to have_selector(:css, "a.btn-danger.disabled")
```

Comment:
Do you think this selector will be a problem in the future if more buttons are added to this partial?  Do other tests scope their selectors to the element in question instead of the whole partial?

Reply:
I'm not sure I completely understand your question.  I will try to answer what I think you are asking, but please clarify if I am answering the wrong question.

 `have_selector` is in common usage in the tests, although most do not use the `:css` as the first parameter.  Many are used to check for the presence of expected buttons., links, fields, etc.

If other buttons/links are added, it is expected they will not have the same `btn-danger` class.

What I'm not sure how to answer is...  
> Do other tests scope their selectors to the element in question instead of the whole partial?

Can you give an example of scoping to the whole partial as opposed to the element?

Reply:
`rendered` here is the partial that is being tested (`hyrax/admin/admin_sets/_show_actions.html.erb`) so I was wondering if it would be good to first find the button element and then make the expectation on that instead of `rendered`.   Maybe something like this https://github.com/samvera/hyrax/blob/master/spec/views/hyrax/citations/work.html.erb_spec.rb#L44.